test/arithmetic/floating_point/floating_point_adder_simple_test.dart

```diff
@@ -10,309 +10,390 @@
+  test('FP: simple adder random', () {
+    const exponentWidth = 5;
+    const mantissaWidth = 20;
+
+    final fp1 = FloatingPoint(
+        exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+    final fp2 = FloatingPoint(
+        exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+    fp1.put(0);
+    fp2.put(0);
     final adder = FloatingPointAdderSimple(fp1, fp2);
+    final value = Random(513);
+    for (var i = 0; i < 500; i++) {
+      final fv1 = FloatingPointValue.random(value,
+          exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+      final fv2 = FloatingPointValue.random(value,
+          exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+      fp1.put(fv1);
+      fp2.put(fv2);
+      final computed = adder.sum.floatingPointValue;
 
-    final fpSuper = adder.sum.floatingPointValue;
-    final fpStr = fpSuper.toDouble().toStringAsPrecision(7);
-    final valStr = out.toDouble().toStringAsPrecision(7);
-    expect(fpStr, valStr);
-  });
-
-  test('FP: small numbers adder test', () {
-    final val = pow(2.0, -23).toDouble();
-    final fp1 = FloatingPoint32()
-      ..put(FloatingPoint32Value.ofDouble(pow(2.0, -23).toDouble()).value);
-    final fp2 = FloatingPoint32()
-      ..put(FloatingPoint32Value.ofDouble(pow(2.0, -23).toDouble()).value);
-    final out = FloatingPoint32Value.ofDouble(val + val);
-
-    final adder = FloatingPointAdderSimple(fp1, fp2);
-
-    final fpSuper = adder.sum.floatingPointValue;
-    final fpStr = fpSuper.toDouble().toStringAsPrecision(7);
-    final valStr = out.toDouble().toStringAsPrecision(7);
-    expect(fpStr, valStr);
-  });
-
-  test('FP: basic loop adder test', () {
-    final input = [(3.25, 1.5), (4.5, 3.75)];
-
-    for (final pair in input) {
-      final fp1 = FloatingPoint32()
-        ..put(FloatingPoint32Value.ofDouble(pair.$1).value);
-      final fp2 = FloatingPoint32()
-        ..put(FloatingPoint32Value.ofDouble(pair.$2).value);
-      final out = FloatingPoint32Value.ofDouble(pair.$1 + pair.$2);
-
-      final adder = FloatingPointAdderSimple(fp1, fp2);
-
-      final fpSuper = adder.sum.floatingPointValue;
-      final fpStr = fpSuper.toDouble().toStringAsPrecision(7);
-      final valStr = out.toDouble().toStringAsPrecision(7);
-      expect(fpStr, valStr);
+      final expectedDouble =
+          fp1.floatingPointValue.toDouble() + fp2.floatingPointValue.toDouble();
+
+      final expectedRound = FloatingPointValue.ofDouble(expectedDouble,
+          exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+
+      final expectedNoRound = FloatingPointValue.ofDoubleUnrounded(
+          expectedDouble,
+          exponentWidth: exponentWidth,
+          mantissaWidth: mantissaWidth);
+
+      if ((computed.mantissa != expectedNoRound.mantissa) &
+          (computed.mantissa != expectedRound.mantissa)) {
+        expect(computed, equals(expectedRound),
+            reason: '\t$fv1 (${fv1.toDouble()})\n'
```

Comment:
you could use a multi-line string for this type of thing, more readable and easier to write
```dart
...
reason: '''
This is
  a multi-line
string'''
```